### ISSUE TITLE
Colors for customers / projects

### ISSUE THREAD

**Issue by npostnik** [kevinpapst/kimai2#374]
It would be great if customers/projects had a color field, the color can be used in time entries view. Then you can see, on which projects/customers you worked. Project color should inherit the customer color if not defined.

**Comment by kevinpapst** [comment]
Hi @npostnik, really like the idea!

Question: I added the option to display customer and project columns next to the already existing activity columns for the timesheet view:
<img width="1492" alt="bildschirmfoto 2018-11-10 um 20 43 41" src="https://user-images.githubusercontent.com/533162/48305351-6f635200-e529-11e8-85f4-7d6ed51757cc.png">

All of the columns can be hidden by the user, as there is not enough room on all devices:

<img width="879" alt="bildschirmfoto 2018-11-10 um 20 53 40" src="https://user-images.githubusercontent.com/533162/48305444-bb62c680-e52a-11e8-9cd4-14e41b0fe47d.png">

Currently anything that is visible is blue, hidden stuff is orange (e.g. projects are hidden if they are directly set to hidden or indirectly through a hidden customer).

Do you have some input how this could be implemented?

Right now I would say: 
- use the customer color for the customer label
- use the project color for the project label

But:
- what about the activity? should it inherit any color? 
- what do we do about the hidden state? 
- should the colors be applied in the admin datatables (customer. project, activity) as well? 
- should the hidden state of the parent be displayed?

The simplest solution if often the best one, but I am especially unsure about the admin datatables and like to get some feedback before starting to work on this one.

**Comment by kevinpapst** [comment]
@maaikez thanks for your feedback and for your PP donation 👍  I'll see what I can do

**Comment by kevinpapst** [comment]
Currently working on this one, as the label cause more trouble than I anticipated in the beginning.
Please give me a quick vote which one you prefer:
**Small dots**
<img width="1151" alt="Bildschirmfoto 2019-05-02 um 17 21 03" src="https://user-images.githubusercontent.com/533162/57086588-b410be80-6cfe-11e9-88b1-d77ec2186441.png">

**Larger dots**
<img width="1175" alt="Bildschirmfoto 2019-05-02 um 17 17 13" src="https://user-images.githubusercontent.com/533162/57086441-67c57e80-6cfe-11e9-896a-faeb67f3fc1b.png">
**Block**
<img width="1182" alt="Bildschirmfoto 2019-05-02 um 17 18 15" src="https://user-images.githubusercontent.com/533162/57086442-67c57e80-6cfe-11e9-9bb9-0ed0c9368ffa.png">
I tend to use the small dots, but let me know what you think please.

**Comment by kevinpapst** [comment]
BTW: invisible items will simply get less opacity, as suggested. I'd say that the information whether something is invisible is not important.

<img width="963" alt="Bildschirmfoto 2019-05-02 um 17 24 37" src="https://user-images.githubusercontent.com/533162/57086979-6ea0c100-6cff-11e9-8dc6-86b7257a3d43.png">

**Comment by maaikez** [comment]
Great you are working on this!

I prefer the blocks.

Are you also adding the colors to the calendar view?

**Comment by kevinpapst** [comment]
I tried it: if the color area gets larger the colors become a quite disturbing "color party" (as you said earlier). So I will stick with the dot for now. 

The calendar will be supported. But there is a small problem: I have to choose a fixed font color. Currently its white: but then you can't use light colors, as they become the background and the font becomes unreadable...

**Comment by kevinpapst** [comment]
You could test the PR if you want.

**Comment by infeeeee** [comment]
Thank you for this, it makes 'My times' much more readable.

But I think the colors should show up on the last activities dropdown as well. The best would be with a colored dot just like on 'my times', but inline. It would make the dropdown consistent with my times, and easier to use.

Should I open a new issue for this?

**Comment by kevinpapst** [comment]
@infeeeee yes, please a new issue. And if possible, include some kind of mockup, because I am not sure if I completely understood how the final outcome should look like.
